modules/core/arrow-test/src/main/kotlin/arrow/test/laws/MonadLaws.kt

```diff
@@ -28,11 +28,14 @@ object MonadLaws {
         Law("Monad Laws: map / flatMap coherence") { M.mapFlatMapCoherence(EQ) },
         Law("Monad Laws: monad comprehensions") { M.monadComprehensions(EQ) },
         Law("Monad Laws: stack safe") { M.stackSafety(5000, EQ) },
-        Law("Monad Laws: selectM == select when Selective has a monad instance") { M.selectEQSelectM(EQ) },
         Law("Monad Laws: monad instance should preserve behavior of Functor") { M.preservesFunctor(FF, EQ) },
         Law("Monad Laws: monad instance should preserve behavior of Applicative") { M.preservesApplicative(AP, EQ) }
       )
 
+  fun <F> laws(M: Monad<F>, FF: Functor<F>, AP: Applicative<F>, SL: Selective<F>, EQ: Eq<Kind<F, Int>>): List<Law> =
+    laws(M, FF, AP, EQ) +
+      Law("Monad Laws: monad instance should preserve behavior of Selective") { M.preservesSelective(SL, EQ) }
+
```

Comment:
Yes this is exactly what I meant, please also move over the tests for applicative and functor to here and revert the original `laws` method back to its previous state. This way the consistency laws are opt-in. We should test them with all datatypes we have in arrow, but as I said in my comment, there are valid reasons to not follow this sometimes.

Reply:
Perfect 👌, although I'll leave usages of this method in tests as they seem to work.

Reply:
If I move functor and applicative to this method we will only be able to test consistency for datatypes which can provide a selective instance. Maybe we can do another overload? We will then have basic monad laws, basic + functor/applicative and basic + functor/ap + selective 🤔

Reply:
Yes we could do another overload, but that will grow in size quite a bit ^-^

Just fill the selective instance in with the monad instance in those cases. The tests, although meaningless, should pass as they are literally the same methods.